src/main/java/com/example/onlinebookshop/services/impl/BookServiceImpl.java

```diff
@@ -70,4 +74,11 @@ public List<BookDto> search(BookSearchParametersDto searchParameters) {
                 .map(bookMapper::toDto)
                 .toList();
     }
+
+    private void validateId(Long id) {
```

Comment:
Just add @Positive Long id in endpoint updateBook and @Validated over the BookController
remove this method